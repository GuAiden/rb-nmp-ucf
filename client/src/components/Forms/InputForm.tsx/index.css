.icon-wrapper {
    background-color: transparent; 
    background-repeat: no-repeat; 
    border: none; 
    outline: none; 
    overflow: hidden; 
    cursor: pointer;
}

.input-modal {
    background-color: grey;
}

.modal-header-wrapper {
    font-size:22px;
    font-style: italic;
    color: white; 
    font-weight: 600;
}

.modal-add-wrapper {
    background-color: transparent; 
    border-color: black; 
}