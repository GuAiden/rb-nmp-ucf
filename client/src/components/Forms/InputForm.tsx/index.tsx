import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';
import CreateInputModal from './createinputmodal';
import gridviewicon from '../../../assets/gridviewicon.png';
import stackedviewicon from '../../../assets/stackedviewicon.png';
import './index.css';

const InputForm: React.FunctionComponent = () => {
  const [state, setState] = useState(0);

  return (
    <React.Fragment>
      <div className="container bg-dark">
        <div className="container bg-dark pt-4">
          <div className="row justify-content-between">
            <div className="col-4 my-auto">
              <CreateInputModal />
            </div>
            <div className="col-4 my-auto">
              <button className="icon-wrapper">
                <img
                  src={gridviewicon}
                  alt="gridView"
                  onClick={(): void => console.log('GridView pressed')}
                />
              </button>
              <button className="icon-wrapper">
                <img
                  src={stackedviewicon}
                  alt="stackedView"
                  onClick={(): void => console.log('StackedView pressed')}
                />
              </button>
            </div>
          </div>
        </div>
        <p>Hello world {state}</p>
        <Button onClick={(): void => setState(1)} />
      </div>
    </React.Fragment>
  );
};

export default InputForm;
