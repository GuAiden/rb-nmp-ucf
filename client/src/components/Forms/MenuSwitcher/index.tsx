import React from 'react';
import './menuswitcher.css';

type MenuSwitcherProps = {
  onFormChange: (form: string) => void;
  form: string;
};

/**
 * Basic anchor group to switch between components
 * @returns Functional component to switch between input/output/server menus
 */
const MenuSwitcher: React.FunctionComponent<MenuSwitcherProps> = ({
  onFormChange,
  form,
}: MenuSwitcherProps) => (
  <div className="container border border-primary mt-5 bg-primary">
    <div className="row">
      <div className="col-sm col-wrapper my-auto">
        <a
          href="#"
          className={form === 'InputForm' ? 'a-selected' : 'a-unselected'}
          onClick={(): void => onFormChange('InputForm')}
        >
          Inputs
        </a>
      </div>
      <div className="col-sm col-wrapper my-auto">
        <hr className="hr-wrapper" />
      </div>
      <div className="col-sm col-wrapper my-auto">
        <a
          href="#"
          className={form === 'OutputForm' ? 'a-selected' : 'a-unselected'}
          onClick={(): void => onFormChange('OutputForm')}
        >
          Outputs
        </a>
      </div>
      <div className="col-sm col-wrapper my-auto">
        <hr className="hr-wrapper" />
      </div>
      <div className="col-sm col-wrapper my-auto">
        <a
          href="#"
          className={form === 'ServerForm' ? 'a-selected' : 'a-unselected'}
          onClick={(): void => onFormChange('ServerForm')}
        >
          Server
        </a>
      </div>
      <div className="col-sm col-wrapper my-auto">
        <hr className="hr-wrapper" />
      </div>
      <div className="col-sm col-wrapper my-auto">
        <a
          href="#"
          className={form === 'SummaryForm' ? 'a-selected' : 'a-unselected'}
          onClick={(): void => onFormChange('SummaryForm')}
        >
          Summary
        </a>
      </div>
    </div>
  </div>
);
export default MenuSwitcher;
