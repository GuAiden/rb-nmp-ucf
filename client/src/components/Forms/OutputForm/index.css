.container-color {
    background-color: rgb(34, 30, 30);
}

.button-clear {
    background-color: transparent;
    border: 2px;
    border-color: white;
}

.stackview-header {
    color: rgb(0, 68, 255);
}

.button-clear {
    background-color: transparent;
    border: 2px;
    border-color: white;
}

#grey {
    background-color: rgb(110, 110, 110);
}

#dark-grey {
    background-color: rgb(53, 53, 53);
}