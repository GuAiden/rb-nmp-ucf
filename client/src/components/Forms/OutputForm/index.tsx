import React from 'react';
import { Container, Row, Col, Button } from 'react-bootstrap';

type OutputFormProps = {
  onFormChange: (form: string) => void;
};

const OutputForm: React.FunctionComponent<OutputFormProps> = ({
  onFormChange,
}: OutputFormProps) => (
  <React.Fragment>
    <Container className="container-color">
      <p className="text-light">Hello Output page</p>
    </Container>
    <Container>
      <Row className="justify-content-between gx-0">
        <Col>
          <Button
            variant="primary"
            className="rounded-0 mt-5 py-2 px-5 float-start button-clear border"
            onClick={(): void => onFormChange('InputForm')}
          >
            <p className="px-2 my-auto button-next">Back</p>
          </Button>
        </Col>
        <Col>
          <Button
            variant="primary"
            className="rounded-0 mt-5 py-2 px-5 float-end"
            onClick={(): void => onFormChange('ServerForm')}
          >
            <p className="px-2 my-auto button-next">Next</p>
          </Button>
        </Col>
      </Row>
    </Container>
  </React.Fragment>
);

export default OutputForm;
