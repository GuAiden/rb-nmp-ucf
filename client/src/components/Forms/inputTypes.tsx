/**
 * Summary output JSON
 */
export interface FirmwareState {
  server: ServerInput;
  inputs: Input[];
  outputs: Output[];
}

export interface ServerInput {
  address: string;
  port: number;
  wsPort: number;
}

export interface Input {
  channelName: string;
  channelNumber: number;
  units: string;
  output: boolean;
  conversion: boolean;
  x?: number;
  y?: number;
}

export interface Output {
  channelName: string;
  channelNumber: number;
  units: string;
  output: boolean;
  conversion: boolean;
  x?: number;
  y?: number;
}
