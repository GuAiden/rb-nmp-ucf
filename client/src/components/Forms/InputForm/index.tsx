import React, { useState } from 'react';
import CreateInputModal from './createinputmodal';
import gridviewicon from '../../../assets/gridviewicon.png';
import stackedviewicon from '../../../assets/stackedviewicon.png';
import './index.css';
import { Input } from '../inputTypes';

type InputFormProps = {
  onInputChange: (userInputs: Input[]) => void;
};

const InputForm = ({ onInputChange }: InputFormProps): JSX.Element => {
  const [inputList, setInputList] = useState<Input[]>([] as Input[]);

  const handleAddInput = (userInput: Input): void => {
    setInputList(inputList.concat(userInput));
    onInputChange(inputList);
  };

  return (
    <React.Fragment>
      <div className="container bg-dark">
        <div className="container bg-dark pt-4">
          <div className="row justify-content-between">
            <div className="col-4 my-auto">
              <CreateInputModal onAddInput={handleAddInput} />
            </div>
            <div className="col-4 my-auto">
              <button className="icon-wrapper">
                <img
                  src={gridviewicon}
                  alt="gridView"
                  onClick={(): void => console.log('GridView pressed')}
                />
              </button>
              <button className="icon-wrapper">
                <img
                  src={stackedviewicon}
                  alt="stackedView"
                  onClick={(): void => console.log('StackedView pressed')}
                />
              </button>
            </div>
          </div>
          <div className="row">
            {inputList.map((d, idx) => (
              <li className="text-light" key={idx}>
                {d.channelName}
              </li>
            ))}
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

export default InputForm;
