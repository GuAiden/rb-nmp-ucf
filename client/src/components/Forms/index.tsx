import React, { useState } from 'react';
import { FirmwareState, Input } from './inputTypes';
import InputForm from './InputForm';

const Forms: React.FunctionComponent = () => {
  const [state, setState] = useState<FirmwareState>({} as FirmwareState);

  const handleInputChange = (userInputs: Input[]): void => {
    setState({ ...state, inputs: userInputs });
    console.log(state.inputs);
  };

  /**
   * Uncomment when components have been made, pass down as prop to relevant component
   */
  // function handleOutputChange(userOutputs: Output[]): void {
  //   setState({ ...state, outputs: userOutputs });
  // }

  // function handleServerChange(inputServer: ServerInput): void {
  //   setState({ ...state, server: inputServer });
  // }

  return (
    <div>
      <InputForm onInputChange={handleInputChange} />
    </div>
  );
};

export default Forms;
