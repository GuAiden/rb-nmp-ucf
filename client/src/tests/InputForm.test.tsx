import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import CreateInputModal from '../components/Forms/InputForm/Create_Input_Modal';
import { Input } from '../components/Forms/Input_Types';

Enzyme.configure({ adapter: new Adapter() });

describe('CreateInputModal component', () => {
  it('renders without failure', () => {
    const handleAddInput = jest.fn();
    const inputListMock = [
      {
        output: false,
        conversion: false,
      },
    ] as Input[];
    const component = shallow(
      <CreateInputModal
        onAddInput={handleAddInput}
        inputList={inputListMock}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
