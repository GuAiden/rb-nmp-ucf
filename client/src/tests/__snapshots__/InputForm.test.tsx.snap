// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateInputModal component renders without failure 1`] = `ShallowWrapper {}`;
